app/actions/remote/entry/common.ts

```diff
@@ -382,6 +392,7 @@ export const syncOtherServers = async (serverUrl: string) => {
                 registerDeviceToken(server.url);
                 syncAllChannelMembersAndThreads(server.url);
                 autoUpdateTimezone(server.url);
+                dataRetentionCleanup(server.url);
```

Comment:
`syncAllChannelMembersAndThreads` may get posts (the root post of the threads) while the `dataRetentionCleanup` may delete post (not sure if also the root post of threads). I am a bit worried this may cause race conditions, since we are not awaiting here for anything.

Reply:
In that case, we can run the cleanup after syncing then? 
```
syncAllChannelMembersAndThreads(server.url).then(() => {
    dataRetentionCleanup(server.url);
});
```

Reply:
LGTM. Just wonder if we should use await instead of then, but 0/5 on that part.